### ISSUE TITLE
Error in applyRemarks in journey-leg.js

### ISSUE THREAD

**Issue by gaudes** [public-transport/hafas-client#196]
Hi,

I get sometimes a error in Function applyRemarks in journey-leg.js.

This is the code where the error occurs:

```
const applyRemarks = (leg, refs) => {
	for (let [remark, ref] of findRemarks(refs)) {
		const {fromLocation, toLocation} = ref

		let fromI = 0, toI = leg.stopovers.length - 1
		// this fails if `s.stop` is a new object (not reference-equal)
		// todo: do this index- or ID-based
		if (fromLocation) {
			fromI = leg.stopovers.findIndex(s => s.stop === fromLocation)
			if (fromI < 0) continue
		}
``` 

Error occurs in line 

```		
let fromI = 0, toI = leg.stopovers.length - 1
```

stopovers in undefined, so length throws an error:

```
TypeError: Cannot read property 'length' of undefined
    at applyRemarks (/opt/iobroker/node_modules/iobroker.fahrplan/node_modules/hafas-client/parse/journey-leg.js:16:38)
    at parseJourneyLeg (/opt/iobroker/node_modules/iobroker.fahrplan/node_modules/hafas-client/parse/journey-leg.js:93:4)
    at Object.args [as parseJourneyLeg] (/opt/iobroker/node_modules/iobroker.fahrplan/node_modules/hafas-client/lib/profile-hooks.js:18:12)
    at j.secL.map.l (/opt/iobroker/node_modules/iobroker.fahrplan/node_modules/hafas-client/parse/journey.js:37:39)
    at Array.map (<anonymous>)
    at parseJourney (/opt/iobroker/node_modules/iobroker.fahrplan/node_modules/hafas-client/parse/journey.js:37:22)
    at Object.args [as parseJourney] (/opt/iobroker/node_modules/iobroker.fahrplan/node_modules/hafas-client/lib/profile-hooks.js:18:12)
    at res.outConL.map.j (/opt/iobroker/node_modules/iobroker.fahrplan/node_modules/hafas-client/index.js:215:22)
    at Array.map (<anonymous>)
    at profile.request.then (/opt/iobroker/node_modules/iobroker.fahrplan/node_modules/hafas-client/index.js:215:5) {stack: 'TypeError: Cannot read property 'length' of u…lan/node_modules/hafas-client/index.js:215:5)', message: 'Cannot read property 'length' of undefined'}
```

This are the values of leg and refs (output of JSON.stringify) for the function when failing:

legs:
```
'{"origin":{"type":"stop","id":"115824","name":"Lotichiusstraße, Schlüchtern","location":{"type":"location","id":"115824","latitude":50.34931,"longitude":9.521599},"products":{"nationalExpress":false,"national":false,"regionalExp":false,"regional":false,"suburban":false,"bus":true,"ferry":false,"subway":false,"tram":false,"taxi":false}},"destination":{"type":"stop","id":"102107","name":"Hallenbad, Schlüchtern","location":{"type":"location","id":"102107","latitude":50.348842,"longitude":9.51999},"products":{"nationalExpress":false,"national":false,"regionalExp":false,"regional":false,"suburban":false,"bus":true,"ferry":false,"subway":false,"tram":false,"taxi":false}},"departure":"2020-11-17T06:12:00+01:00","plannedDeparture":"2020-11-17T06:12:00+01:00","departureDelay":null,"arrival":"2020-11-17T06:15:00+01:00","plannedArrival":"2020-11-17T06:15:00+01:00","arrivalDelay":null,"public":true,"walking":true,"distance":126}'
```

refs:
```
'[{"type":"REM","remX":5,"tagL":["RES_SEC_FTR_H3"],"hint":{"type":"hint","code":"X8","text":"Weg 100 m"}}]'
```

**Comment by derhuerst** [comment]
This should be fixed in 66ff661767a5622a9e942774ec6108c6500e26b3, published as `hafas-client@5.10.1`. Can you confirm that?

**Comment by gaudes** [comment]
Hi @derhuerst,

I use this module in an adapter for ioBroker.
I found this problem via Sentry error reporting from installations "in the wild".

I've done a short local test. Error occured with a specific journey, quick update to 5.10.1, error was gone.

Seems to fix the problem :)

Do you know when this version will be available via NPM ?

Thank you very much for your fast reaction.

Regards, Ralf

**Comment by derhuerst** [comment]
> Do you know when this version will be available via NPM ?

Shit, totally forgot to publish to npm. I have published `5.10.1` to npm now.
